main.cpp

```diff
@@ -311,13 +222,20 @@ struct DoubleType
     DoubleType& multiply(double rhs);
     DoubleType& divide(double rhs);
 
+    DoubleType& pow(double rhs);
+    DoubleType& pow(const FloatType& rhs);
+    DoubleType& pow(const DoubleType& rhs);
+    DoubleType& pow(const IntType& rhs);
+
 private:
+    DoubleType& powInternal(double* base, double exp);
```

Comment:
The first argument is not needed.
```
 2) your powInternal() function should do something like this in its body:    *val = std::pow( *val, arg );
```
the internal `val/value` pointer is always used inside this function, and does not need to be passed as a function argument
- [ ] revise